common/persistence/dataManagerInterfaces_test.go

```diff
@@ -58,3 +60,65 @@ func TestIsTransientError(t *testing.T) {
 		require.False(t, IsTransientError(err))
 	}
 }
+
+func TestIsTimeoutError(t *testing.T) {
+	notTimeoutError := fmt.Errorf("not timeout error")
+	assert.False(t, IsTimeoutError(notTimeoutError))
+	assert.True(t, IsTimeoutError(&TimeoutError{}))
+}
+
+func TestTaskCommonMethods(t *testing.T) {
+	timeNow := time.Now()
+	tasks := []interface{}{
+		&ActivityTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&DecisionTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&RecordWorkflowStartedTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&ResetWorkflowTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&CloseExecutionTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&DeleteHistoryEventTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&DecisionTimeoutTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&ActivityTimeoutTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+		&UserTimerTask{Version: 1, TaskID: 1, VisibilityTimestamp: timeNow},
+	}
+
+	for _, task := range tasks {
```

Comment:
you can define `tasks` as a map to co-locate object and its expected type

Reply:
Given that transfer task and non-transfer task share the same set of int keys, I think we should keep the current way of testing.